core/templates/domain/exploration/ExplorationMetadataObjectFactory.ts

```diff
@@ -0,0 +1,140 @@
+// Copyright 2022 The Oppia Authors. All Rights Reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS-IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+/**
+ * @fileoverview Factory for creating instances of exploration metadata objects.
```

Comment:
Can we create a model instead of an object factory? We seem to prefer models per https://github.com/oppia/oppia/issues/10700.

Reply:
I think that for now, it should be an object factory because the function `createFromBackendDict` involves use of `ParamChangesObjectFactory` and `ParamSpecsObjectFactory`. Hence, these two would need to be migrated first before we can migrate the ExplorationMetadataObjectFactory into a model.

Reply:
I will migrate this into a model after `ParamChangesObjectFactory` and `ParamSpecsObjectFactory` are migrated.

Reply:
Should we add this file to #10700? And perhaps add a TODO in this PR.

/cc @srijanreddy98 as FYI

Reply:
Done.